pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -383,11 +383,20 @@ stdenv.mkDerivation ({
       mv tmp "$d"
     done
 
-    for d in $(grep '^dynamic-library-dirs:' "$packageConfDir"/* | cut -d' ' -f2- | tr ' ' '\n' | sort -u); do
-      for lib in "$d/"*.{dylib,so}; do
-        ln -s "$lib" "$dynamicLinksDir"
+    listDynamicLinks () {
```

Comment:
Could you add a comment on this function explaining what it is doing (and why it is needed)?